client/src/selectors.js

```diff
@@ -0,0 +1,56 @@
+// @flow
+import {createSelector} from 'reselect'
+import {sortBy, chunk} from 'lodash'
+import queryString from 'query-string'
+import {paginationChunkSize} from './constants'
+import {values} from './utils'
+
+import type {Location} from 'react-router-dom'
+import type {State} from './state'
+import type {NoticesOrdering} from './components/NoticeList'
+import type {NoticeDetailProps} from './components/NoticeDetail'
+
+export const noticeDetailSelector = (state: State, props: NoticeDetailProps) =>
+  props.match.params.id ? state.notices.data[props.match.params.id] : undefined
```

Comment:
props.match.params.id && state.notices.data[props.match.params.id]

Reply:
not the same, flow would complain

Reply:
sad parrot :( but I see

Reply:
I take it back, beforehand I did not have correct definitions for match.params, works fine now - fixed